apollo-federation/src/connectors/validation/snapshots/validation_tests@env-vars.graphql.snap

```diff
@@ -13,7 +13,7 @@ input_file: apollo-federation/src/connectors/validation/test_data/env-vars.graph
     },
     Message {
         code: InvalidHeader,
-        message: "In `@source(http.headers:)`: `$env.foo` doesn't have a field named `bar`",
+        message: "In `@source(http.headers:)`: `$env.foo` doesn't have a field named `.bar`",
```

Comment:
The addition of these `.` characters is an expected and hopefully harmless consequence of making each `NamedShapePathKey` variant handle printing its own `.` in the `Display` implementation. The new `?` and `!` path elements should not be preceded by a `.`, whereas the existing `.*`, `.**`, `.field`, and `.3` keys all need `.`s. Previously, the `.` was added at a higher level for all variants, but now that not all variants need it, the ones that do will print like `.bar` above. If this seems important to avoid, we could consider stripping the leading `.` when we construct the error message.